desci-server/src/routes/v1/attestations/schema.ts

```diff
@@ -71,24 +75,39 @@ const highlightSchema = z
       highlight.pageIndex !== null &&
       highlight.pageIndex !== undefined &&
       (highlight.text || highlight.image),
+    { message: 'Invalid Pdf highlight block' },
   );
 
+const codeHighlightSchema = z.object({
+  id: z.string(),
+  text: z.string().optional(),
+  path: dpidPathSchema,
+  cid: z.string(),
+  startLine: z.coerce.number(),
+  endLine: z.coerce.number(),
+  language: z.string(),
+  kind: z.literal('code'),
+});
```

Comment:
IMO, having these hard-coded schemas is confusing. It's been quite tricky to keep them in sync between desci-models and desci-repo for example. We should look into generating zod validators from the types themselves, instead of defining them separately.

Reply:
when we move all social features to ceramic these models will be published through desci-models

Reply:
Things will be more stable by then but for now I can't because it'll slow down development (having to publish new model versions to make quick changes lol)